ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java

```diff
@@ -1655,6 +1655,12 @@ public boolean alreadyExists(PersistentIndex idx, FileSpan searchSpan, StoreKey
             }
           }
         } else if (currentValue.isTtlUpdate()) {
+          if (currentLatestState == null) {
+            // Only one case this would happen, an TTL_UPDATE indexValue that has no PUT before, this is because the put
```

Comment:
`an TTL_UPDATE indexValue that has no PUT before and no DELETE at the end, this is because ...`